lib/PrintAsClang/ModuleContentsWriter.cpp

```diff
@@ -457,6 +457,14 @@ class ModuleWriter {
       }
     }
 
+    if (isa<EnumDecl>(D) && !D->hasClangNode() &&
+        outputLangMode != OutputLanguageMode::Cxx) {
```

Comment:
I am not sure I follow from the code, could you explain why `outputLangMode != OutputLanguageMode::Cxx`?